client/components/jetpack/backup-schedule-setting/index.tsx

```diff
@@ -1,39 +1,118 @@
-		options.push( {
-			label: `${ startTime } - ${ endTime }`,
-			value: hour.toString(),
-		} );
-	}
-	return options;
-};
-
 const BackupScheduleSetting: FunctionComponent = () => {
+	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const options = generateTimeSlots();
+	const queryClient = useQueryClient();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const timezone = useSelector( ( state ) => getSiteTimezoneValue( state, siteId ) );
+	const gmtOffset = useSelector( ( state ) => getSiteGmtOffset( state, siteId ) );
+
+	const convertHourToRange = ( hour: number ): string => {
+		const period = hour < 12 ? 'AM' : 'PM';
+		const adjustedHour = hour % 12 || 12;
+		return `${ adjustedHour }:00 - ${ adjustedHour }:59 ${ period }`;
+	};
+
+	const generateTimeSlots = (): { label: string; value: string }[] => {
+		const options = [];
+		for ( let hour = 0; hour < 24; hour++ ) {
+			const utcTime = moment.utc().startOf( 'day' ).hour( hour );
+			const localTime =
+				timezone && gmtOffset
+					? applySiteOffset( utcTime, { timezone, gmtOffset } )
+					: utcTime.local();
+			const localHour = localTime.hour();
+			const timeRange = convertHourToRange( localHour );
+
+			options.push( {
+				label: timeRange,
+				value: hour.toString(),
+				localHour, // for sorting
+			} );
+		}
+
+		// Sort options by local hour before returning
+		options.sort( ( a, b ) => a.localHour - b.localHour );
+
+		// Remove the localHour from the final result as it's not needed anymore
+		return options.map( ( { label, value } ) => ( { label, value } ) );
+	};
+
+	const timeSlotOptions = generateTimeSlots();
+	const { isFetching: isScheduledTimeQueryFetching, data } = useScheduledTimeQuery( siteId );
+	const { isPending: isScheduledTimeMutationLoading, mutate: scheduledTimeMutate } =
+		useScheduledTimeMutation( {
+			onSuccess: () => {
+				queryClient.invalidateQueries( { queryKey: [ 'jetpack-backup-scheduled-time', siteId ] } );
+				dispatch(
+					successNotice( translate( 'Daily backup time successfully changed.' ), {
+						duration: 5000,
+						isPersistent: true,
+					} )
+				);
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'Update daily backup time failed. Please, try again.' ), {
+						duration: 5000,
+						isPersistent: true,
+					} )
+				);
+			},
+		} );
+
+	const isLoading = isScheduledTimeQueryFetching || isScheduledTimeMutationLoading;
+
+	const updateScheduledTime = ( selectedTime: string ) => {
+		scheduledTimeMutate( { scheduledHour: Number( selectedTime ) } );
+	};
+
+	const getScheduleInfoMessage = (): TranslateResult => {
+		const hour = data?.scheduledHour || 0;
+		const range = convertHourToRange( hour );
+
+		if ( ! data || ! data.scheduledBy ) {
+			return `${ translate( 'Default time' ) }. UTC: ${ range }`;
+		}
+		return `${ translate( 'Time set by %(scheduledBy)s', {
+			args: { scheduledBy: data.scheduledBy },
+		} ) }. UTC: ${ range }`;
```

Comment:
In the case of UTC, there is no AM/PM, it uses 24-hour clock

Reply:
Oh, TIL. Fixing it!